android/cli/commands/_build.js

```diff
@@ -3600,14 +3626,26 @@ AndroidBuilder.prototype.generateAndroidManifest = function generateAndroidManif
 	tiappServices && Object.keys(tiappServices).forEach(function (filename) {
 		const service = tiappServices[filename];
 		if (service.url) {
-			const s = {
-				'name': this.appid + '.' + service.classname
-			};
-			Object.keys(service).forEach(function (key) {
-				if (!/^(type|name|url|options|classname|android:name)$/.test(key)) {
-					s[key.replace(/^android:/, '')] = service[key];
-				}
-			});
+			const s = {};
+			if (service.type === 'quicksettings') {
+				const serviceName = this.appid + '.' + service.classname;
+				const icon = '@drawable/' + (service.label || this.tiapp).icon.replace(/((\.9)?\.(png|jpg))$/, '');
+				const label = service.label || this.tiapp.name;
+				const serviceXML = ejs.render(fs.readFileSync(path.join(this.templatesDir, 'QuickService.xml')).toString(), {
+					serviceName: serviceName,
+					icon: icon,
+					label: label
+				});
+				const doc = new DOMParser().parseFromString(serviceXML, 'text/xml');
+				serviceParser(doc.firstChild, s);
```

Comment:
How about serviceParser doesn't get passed in `s`, since it's an empty object, and instead just creates an empty object to start from and returns the built object, with we then assign to `s` here?
`s = serviceParser(doc.firstChild);`

It would mean that line 3629 would need to change to `let s = {};`.